socorro/external/es/crashstorage.py

```diff
@@ -138,22 +143,91 @@ def _submit_crash_to_elasticsearch(self, connection, crash_document):
             index_creator.create_socorro_index(es_index)
 
         # Submit the crash for indexing.
-        try:
-            connection.index(
-                index=es_index,
-                doc_type=es_doctype,
-                body=crash_document,
-                id=crash_id
-            )
+        success = False
+        # Don't retry more than 5 times. That is to avoid infinite loops in
+        # case of an unhandled exception.
+        retries_pool = [True] * 5 + [False]
+        while not success and retries_pool.pop(0):
+            try:
+                connection.index(
+                    index=es_index,
+                    doc_type=es_doctype,
+                    body=crash_document,
+                    id=crash_id
+                )
+                success = True
+            except elasticsearch.exceptions.TransportError as e:
+                field_name = None
+
+                if 'MaxBytesLengthExceededException' in e.error:
+                    # This is caused by a string that is way too long for
+                    # Elasticsearch.
+                    matches = re.findall(
+                        FIELD_NAME_STRING_ERROR_REGEX, e.error
+                    )
+                    if matches:
+                        field_name = matches[0]
+                elif 'NumberFormatException' in e.error:
+                    # This is caused by a number that is either too big for
+                    # Elasticsearch or just not a number.
+                    matches = re.findall(
+                        FIELD_NAME_NUMBER_ERROR_REGEX, e.error
+                    )
+                    if matches:
+                        field_name = matches[0]
+                else:
+                    # We don't know how to fix this exception. Let it raise.
+                    self.config.logger.critical(
+                        'Submission to Elasticsearch failed for %s (%s)',
+                        crash_id,
+                        e,
+                        exc_info=True
+                    )
+                    raise
 
-        except elasticsearch.exceptions.ElasticsearchException as e:
-            self.config.logger.critical(
-                'Submission to Elasticsearch failed for %s (%s)',
-                crash_id,
-                e,
-                exc_info=True
-            )
-            raise
+                if not field_name:
+                    # We are unable to parse which field to remove, we cannot
+                    # try to fix the document. Let it raise.
+                    self.config.logger.critical(
+                        'Submission to Elasticsearch failed for %s (%s)',
+                        crash_id,
+                        e,
+                        exc_info=True
+                    )
+                    raise
```

Comment:
Perhaps there's something I'm missing but isn't this repeated code?
Couldn't you just do...
```python
except elasticsearch.exceptions.TransportError as e:
    field_name = None
    if 'MaxBytesLengthExceededException' in e.error:
        ....
    elif 'NumberFormatException' in e.error:
        ...
    if not field_name:
        self.config.logger.critical(...)
        raise
```